### DISCUSSION TITLE
What's the equivalent to useControlledState in v8?

### DISCUSSION THREAD

**Discussion by em2724** [TanStack/table#5327] [Q&A]
I store selected ids in the url and want to update the table rowSelection state whenever that value changes. Previously useControlledState worked well for this where you are passed the oldState and return the newState.

What's the equivalent to useControlledState in v8? Currently I use a useEffect like this:

```
React.useEffect(() => {
    table.setState(old => {
      return { ...old, rowSelection: // update based on searchParams };
    });
  }, [searchParams]);
```

Is this the intention? Is there a better way?
Thanks.

**Comment by goatfryed** [comment]
To control state in v8, you provide your specific state slice in the options and a general or feature specific change listener. In case of rowSelection, this is [onRowSelectionChange](https://tanstack.com/table/v8/docs/api/features/row-selection#onrowselectionchange)

Also see the example on [fully-controlled state](https://tanstack.com/table/v8/docs/framework/react/examples/fully-controlled)
